Need to update renewIssueEtc when renewing cookie

HaoK opened the pull request:
cc @Tratcher

dnfclas commented:
Hi __@HaoK__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

HaoK commented:
Fixes https://github.com/aspnet/Security/issues/628

I'm not sure if we need to handle sliding expiry as well here...

Tratcher commented:
I think you need to handle sliding refresh.

https://github.com/aspnet/Security/blob/haok/1-6-issue/src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs#L85-L86
If they access the site every 30 seconds and issuedUtc is updated each time, then timeElapsed will always be 30s and timeRemaining will keep shrinking. The sliding expiration won't be renewed until you're within 30s, and if you miss a request in that window then your cookie suddenly expires.

HaoK commented:
Yeah I thought as much, this cookie refresh/expires code is pretty complicated as its written :/

HaoK commented:
I think I will try to refactor the cookie refresh/renew logic into some sane methods so this is actually readable and sane...

HaoK commented:
Updated:
- Moved logic for refresh into two helpers
- We request refresh if explicitly asked for via context.ShouldRenew, or sliding check
- If we renew, we ALWAYS set expires/issues UTC (is that correct/safe?)

Tratcher commented on src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs:
```diff
@@ -37,6 +37,33 @@ private Task<AuthenticateResult> EnsureCookieTicket()
+
+        private void RequestRefresh(AuthenticationTicket ticket)
+        {
+            _shouldRenew = true;
```
Any idea why the variables all say renew and not refresh? Are they used for anything else?

HaoK replied:
Not used anywhere else I don't think, I'll rename to refresh, seems fine

Tratcher commented:
:shipit: and consider updating the naming.

HaoK commented:
5bcc1bfc260978b0bea9b981056dd0beb31df400
